ServerMonitoringSystem.MessageProcessor/Program.cs

```diff
@@ -1,85 +1,138 @@
 ﻿using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
-using ServerMonitoringSystem.MessageProcessor.Configuration;
-using ServerMonitoringSystem.MessageProcessor.Services;
-using ServerMonitoringSystem.MessageProcessor.Services.Interfaces;
 using MessagingLibrary.Interfaces;
 using MessagingLibrary.RabbitMq;
-using ServerMonitoringSystem.Shared.Configuration;
-using ServerMonitoringSystem.Infrastructure.Settings;
 using ServerMonitoringSystem.Infrastructure.Repositories;
+using ServerMonitoringSystem.Infrastructure.Settings;
+using ServerMonitoringSystem.MessageProcessor.Configuration;
 using ServerMonitoringSystem.MessageProcessor.Persistence;
+using ServerMonitoringSystem.MessageProcessor.Services;
+using ServerMonitoringSystem.MessageProcessor.Services.Interfaces;
+using ServerMonitoringSystem.Shared.Configuration;
 
-var builder = new ConfigurationBuilder()
+var configuration = new ConfigurationBuilder()
     .SetBasePath(Directory.GetCurrentDirectory())
-    .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true);
-
-var configuration = builder.Build();
+    .AddJsonFile("appsettings.json", optional: true)
+    .AddEnvironmentVariables()
+    .Build();
 
-var anomalyConfig = configuration
-    .GetSection("AnomalyDetectionConfig")
-    .Get<AnomalyDetectionConfig>();
-
-var signalRConfig = configuration
-    .GetSection("SignalRConfig")
-    .Get<SignalRConfig>();
-
-var mongoDbSettings = configuration
-    .GetSection("MongoDbSettings")
-    .Get<MongoDbSettings>();
-
-var serverConfig = new ServerStatisticsConfig
+try
 {
-    ServerIdentifier = "Server1" 
-};
-
-var services = new ServiceCollection();
-services.AddSingleton<IConfiguration>(configuration);
-services.AddSingleton(anomalyConfig);
-services.AddSingleton(signalRConfig);
-services.AddSingleton(serverConfig);
-services.AddSingleton(mongoDbSettings);
-services.AddScoped<IStatisticsRepository, MongoDbStatisticsRepository>();
-
-
-string rabbitMqHost = "localhost";
-string exchange = "ServerExchange";
-string queue = "ServerStatsQueue";
-string routingKey = "ServerStatistics.*";
-services.AddSingleton<IMessageConsumer>(provider =>
+    var anomalyConfig = new AnomalyDetectionConfig
+    {
+        MemoryUsageAnomalyThresholdPercentage = double.Parse(
```

Comment:
we notice that we have lots of duplication in `Program.cs` files where we try to read from appsettings, if value is not there, you read then from `.env`. Lets make a generic solution to read the configs :D FYI there's a way to read appsettings and load a section into a model.